qemu/tests/virtio_trace_pipenb.py

```diff
@@ -0,0 +1,41 @@
+import logging
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Under named-pipe non-blocking testing:
+    1) Create pipe named by the following
+    2) Boot up a single-CPU guest with a virtio-serial device and
+       named-pipe chardev backend
+    3) Run a background work on the guest.
+    4) Write data to the virtio-serial port until the guest stops.
+    5) Read the named-pipe file on the host.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def execute_host_cmd(host_cmd, timeout=60):
+        """
+        Execute the host_cmd on host, limited in timeout period
+
+        param host_cmd: the host_cmd to run on host
+        param timeout: the timeout for running this command
+        return: the output of the host_cmd
+        """
+        logging.info("Executing host command: %s" % host_cmd)
+        cmd_result = process.getstatusoutput(host_cmd, timeout=timeout, shell=True)
+        return cmd_result
+
+    timeout = float(params.get("login_timeout", 240))
+    vm = env.get_vm(params["main_vm"])
+    v_path = vm.get_serial_console_filename('vs2')
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    session.cmd_status_output(" while :;do sleep 1; date; done &")
+    session.cmd_status_output("cat /proc/kallsyms > /dev/virtio-ports/vs2 &")
+    execute_host_cmd("cat {}".format(v_path))
```

Comment:
No need, only use one time, use `process` directly is ok.

Reply:
Only read the named-pipe file on the host, but no checking. The checking step is what this case want to test.
```
The data could be read correctly
Confirm that the guest resumes its execution
```